### PR TITLE
Remove all uses of Tag from climada/engine

### PR DESCRIPTION
**Pull request by peanutfun**
Changes proposed in this PR:
- Remove all uses of `Tag` from `climada/engine/`

### PR Author Checklist

- [x] Read the [Contribution Guide][contrib]
- [x] Correct target branch selected (if unsure, select `develop`)
- [x] Descriptive pull request title added
- [x] Source branch up-to-date with target branch
- [x] [Documentation](https://climada-python.readthedocs.io/en/latest/guide/Guide_PythonDos-n-Donts.html#2.--Commenting-&-Documenting) updated
- [x] [Tests][testing] updated
- [x] [Tests][testing] passing
- [x] No new [linter issues][linter]
- [x] [Changelog](https://github.com/CLIMADA-project/climada_python/blob/main/CHANGELOG.md) updated

### PR Reviewer Checklist

- [ ] Read the [Contribution Guide][contrib]
- [ ] [CLIMADA Reviewer Checklist](https://climada-python.readthedocs.io/en/latest/guide/Guide_Reviewer_Checklist.html) passed
- [ ] [Tests][testing] passing
- [ ] No new [linter issues][linter]

[contrib]: https://github.com/CLIMADA-project/climada_python/blob/main/CONTRIBUTING.md
[testing]: https://climada-python.readthedocs.io/en/latest/guide/Guide_Continuous_Integration_and_Testing.html
[linter]: https://climada-python.readthedocs.io/en/stable/guide/Guide_Continuous_Integration_and_Testing.html#3.C.--Static-Code-Analysis

### CODE REVIEW

File: climada/engine/impact.py [hunk 1]
```diff
@@ -1215,20 +1171,6 @@ def from_hdf5(cls, file_path: Union[str, Path]):
             ├─ event_name
             ├─ frequency
             ├─ imp_mat
```
Comment (chahank) [inline climada/engine/impact.py:1173]: I think the `haz_type` is now missing from the description docstring. Should be in the `.attrs` ?

File: climada/engine/impact.py [hunk 2]
```diff
@@ -1719,7 +1652,7 @@ def concat(cls, imp_list: Iterable, reset_event_ids: bool = False):
           ``frequency``, ``imp_mat``, ``at_event``,
         - sums up the values of attributes ``eai_exp``, ``aai_exp``
         - and takes the following attributes from the first impact object in the passed
-          impact list: ``coord_exp``, ``crs``, ``unit``, ``tot_value``, ``tag``,
+          impact list: ``coord_exp``, ``crs``, ``unit``, ``tot_value``,
           ``frequency_unit``
```
Comment (chahank) [inline climada/engine/impact.py:1656]:
```suggestion
          impact list: ``coord_exp``, ``crs``, ``unit``, ``tot_value``,
          ``frequency_unit``, ``haz_type``
```

File: climada/util/yearsets.py [hunk 3]
```diff
@@ -82,7 +82,6 @@ def impact_yearset(imp, sampled_years, lam=None, correction_fac=True, seed=None)
 
     #save calculations in yimp
     yimp.event_id = np.arange(1, n_sampled_years+1)
-    yimp.tag['yimp object'] = True
```
Comment (chahank) [inline climada/util/yearsets.py:85]: Does this potentially break a functionality? This is one of the rare instances of a possibly relevant variable to be stored in the tags object.

**Review by chahank — CHANGES_REQUESTED** [review]
Great progress on the quest towards removing the `Tag` from CLIMADA. Thanks!

File: climada/engine/impact.py (hunk 1 above)
Comment (peanutfun) [reply climada/engine/impact.py:1173]: True, looks like we are not writing it anymore. Will fix!

File: climada/util/yearsets.py (hunk 3 above)
Comment (peanutfun) [reply climada/util/yearsets.py:85]: Not sure 😅 All tests pass on my end, so I don't think it does? 🤷

File: climada/engine/impact.py (hunk 1 above)
Comment (peanutfun) [reply climada/engine/impact.py:1173]: It was written, but the test was not adapted properly. Everything works now!

**Comment by peanutfun** [comment]
@chahank There is another issue: `Impact.from_eih` now has an unused argument `impfset`: https://github.com/CLIMADA-project/climada_python/pull/743/files#diff-1c276908f04d1a0ca5b1692295a586ce54b0937673711203582d3d72099ecdd0R210

The `impfset` was only used to populate the tag. I propose to deal with this the following way:
* Remove the `impfset` parameter from the method immediately
* Update all uses of `from_eih` within Climada (there are not many)
* Add a type check for the second argument. If it is an `ImpactFuncSet`, raise an error with useful error message
This way, we are *not* backward compatible but users know how to fix the issue.

**Comment by chahank** [comment]
> @chahank There is another issue: `Impact.from_eih` now has an unused argument `impfset`: https://github.com/CLIMADA-project/climada_python/pull/743/files#diff-1c276908f04d1a0ca5b1692295a586ce54b0937673711203582d3d72099ecdd0R210
> 
> The `impfset` was only used to populate the tag. I propose to deal with this the following way:
> 
>     * Remove the `impfset` parameter from the method immediately
> 
>     * Update all uses of `from_eih` within Climada (there are not many)
> 
>     * Add a type check for the second argument. If it is an `ImpactFuncSet`, raise an error with useful error message
> 
> 
> This way, we are _not_ backward compatible but users know how to fix the issue.

Excellent idea! Although I think we do not need to make the error message. Most users will not directly use this method anyway imho.

**Comment by chahank** [comment]
Oh, and probably you should go through the documentation and check whether the `tag` is explicitly mentioned. If yes, please remove it.

**Comment by peanutfun** [comment]
@chahank I removed the `impfset` parameter from `Impact.from_eih` without adding an type check, as discussed. I removed the mention of Tag from the impact tutorial. It is still mentioned in other tutorials (e.g. exposures and hazard) though, but I would remove these instances once tags are removed from the respective classes.

File: climada/util/yearsets.py (hunk 3 above)
Comment (carmensteinmann) [reply climada/util/yearsets.py:85]: I quickly checked and don't think it would break any functionality.

**Comment by chahank** [comment]
Excellent, looks good to me!

**Comment by peanutfun** [comment]
Thanks for the input, merging now!

**Pull request merged**